packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -2270,11 +2270,14 @@ class VisualDensity with Diagnosticable {
 
   /// Return a copy of [constraints] whose minimum width and height have been
   /// updated with the [baseSizeAdjustment].
+  ///
+  /// The resulting minWidth and minHeight values are clamped to not exceed the
+  /// maxWidth and maxHeight values, respectively.
   BoxConstraints effectiveConstraints(BoxConstraints constraints){
     assert(constraints != null && constraints.debugAssertIsValid());
     return constraints.copyWith(
```

Comment:
It looks likes this should be:
```dart
return constraints.constrain(constraints.smallest + baseSizeAdjustment);
```

What's the difference?

Reply:
The difference is that constrain returns a `Size`, and we need to return updated constraints.

Reply:
Sorry, I wasn't paying attention. The idea was to write an expression where it was obvious that we were applying the existing constraints to the adjusted version.  There doesn't seem to be an obvious way to do that with the BoxConstraints methods.